nautobot/extras/models/jobs.py

```diff
@@ -799,87 +815,150 @@ def enqueue_job(
-
-                # Set alarm_handler to be called on a SIGALRM, and schedule a SIGALRM based on the soft time limit
-                signal.signal(signal.SIGALRM, alarm_handler)
-                signal.alarm(int(job_model.soft_time_limit) or settings.CELERY_TASK_SOFT_TIME_LIMIT)
-
-                try:
-                    eager_result = run_job.apply(
+        job_queue = JobQueue.objects.get(name=task_queue)
+        # Kubernetes Job Queue logic
+        if job_queue.queue_type == JobQueueTypeChoices.TYPE_KUBERNETES:
+            config_file_path = "./development/kind-kube-config"
+            config.load_kube_config(config_file_path)
+            core_v1 = core_v1_api.CoreV1Api()
+            api_instance = core_v1
+            name = "nautobot-pod-demo"
+            resp = None
+            try:
+                resp = api_instance.read_namespaced_pod(name=name, namespace="default")
+                print("Found old pod")
+            except ApiException as e:
+                if e.status != 404:
+                    print(f"Unknown error: {e}")
+                    sys.exit(1)
+
+            if not resp:
+                print(f"Pod {name} does not exist. Creating it...")
+                pod_manifest = {
+                    "apiVersion": "v1",
+                    "kind": "Pod",
+                    "metadata": {"name": name},
+                    "spec": {
+                        "containers": [
+                            {
+                                "image": "networktocode/nautobot:latest",
+                                "name": "nautobot",
+                                "ports": [{"containerPort": 8000}],
+                            }
+                        ]
+                    },
+                }
+                resp = api_instance.create_namespaced_pod(body=pod_manifest, namespace="default")
+                print("create pod")
+                while True:
+                    resp = api_instance.read_namespaced_pod(name=name, namespace="default")
+                    if resp.status.phase != "Pending":
+                        break
+                    time.sleep(1)
+                print("Done.")
+
+            # Calling exec interactively
+            exec_command = ["/bin/sh"]
+            resp = stream(
+                api_instance.connect_get_namespaced_pod_exec,
+                name,
+                "default",
+                command=exec_command,
+                stderr=True,
+                stdin=True,
+                stdout=True,
+                tty=False,
+                _preload_content=False,
+            )
+            resp.write_stdin("nautobot-server runjob --local -u admin nautobot.core.jobs.ExportObjectList\n")
+            sresult = resp.read_stdout()
+            serror = resp.read_stderr()
+            print(f"Job Result is: {sresult}")
+            print(f"Error is: {serror}")
+            resp.close()
+            return job_result
+        else:
```

Comment:
Since we're `return`-ing above, the `else` here and corresponding indentation change isn't *strictly* necessary.